src/Cli/Microsoft.TemplateEngine.Cli/Commands/InteractiveMode/InteractiveMode.cs

```diff
@@ -0,0 +1,253 @@
+                    {
+                        return;
+                    }
+
+                    InteractiveQuerying queries = new InteractiveQuerying();
+                    TemplateCommand? templateCommand = GetTemplate(
+                        knownArgs,
+                        settings,
+                        templateGroups,
+                        templatePackageManager);
+
+                    if (templateCommand is null)
+                    {
+                        return;
+                    }
+
+                    context.Console.WriteLine(InteractiveModePrompts.OpeningMessage("<Default Language>", templateCommand.Template.Name, "skip button"));
+
+                    queries.SetQuestions(knownArgs, templateCommand);
+
+                    var questionsToAsk = queries.Questions();
+                    // TODO: adjust to different input types
+                    Dictionary<string, string> paramsToAdd = new Dictionary<string, string>();
+
+                    while (questionsToAsk.MoveNext())
+                    {
+                        var currentQuestion = questionsToAsk.Current;
+                        bool rightInfoCollected = false;
+                        while (!rightInfoCollected)
+                        {
+                            context.Console.WriteLine(currentQuestion.GetQuery());
+                            var paramValue = Console.ReadLine();
+                            if (paramValue is not null
+                                && paramValue != string.Empty
+                                && paramValue.GetType() == currentQuestion.GetValueType())
+                            {
+                                // only supporting string for now
+                                paramsToAdd.Add(currentQuestion.GetParameterName(), paramValue);
+                                rightInfoCollected = true;
+                                //TODO: Necessary checks to see if value by customer is good
+                            }
+                            if (paramValue == string.Empty)
+                            {
+                                rightInfoCollected = true;
+                            }
+                        }
+                    }
+
+                    // Hard coded for now bc I cannot find where this information would be in the parser / parserResult
+                    // Need to get input given by user
+                    string commandLineCall = $"dotnet new {templateCommand.Template.ShortNameList[0]}";
+                    foreach (var param in paramsToAdd)
+                    {
+                        commandLineCall += ($" --{param.Key} {param.Value}");
+                    }
+                    // After everything is done reparse the command line input and execute the result
+
+                    Parser parser = ParserFactory.CreateParser(context.ParseResult.RootCommandResult.Command);
+                    parser.Invoke(commandLineCall);
```

Comment:
Will this way trigger the same middleware again?

Reply:
If we pass the interactive mode options, and the environmental flag than yes. But I think we might be able to disable both at this point if necessary.